config/functions

```diff
@@ -1035,7 +1043,9 @@ source_package() {
       fi
     fi
 
-    PKG_BUILD="$BUILD/${PKG_NAME}-${PKG_VERSION}"
+    PKG_BUILD="$BUILD/build/${PKG_NAME}-${PKG_VERSION}"
+    PKG_INSTALL="$BUILD/install_pkg/${PKG_NAME}-${PKG_VERSION}"
+    PKG_INSTALL_INIT="$BUILD/install_init/${PKG_NAME}-${PKG_VERSION}"
```

Comment:
Having both `PKG_INSTALL` and `PKG_INSTALL_INIT` feels wrong, and unnecessary.

If the package being sourced is `xyz:init` then `PKG_INSTALL` is `$BUILD/install_init/...`, and if `xyz:target` (or just `xyz`) is sourced then it's `$BUILD/install_pkg/...` - anything else (ie. `:bootstrap` or `:host`) then it's undefined/not set. This would require adding some "target" (`:host`, `:target`, etc.) detection logic to `source_package()` but that should be possible (although be aware it is possible to source a package using the path to the `package.mk` rather than just a package name - with optional target - in which case assume `:target`).

As this is basically the same logic we have in `scripts/build` and `scripts/install`, the logic in both these scripts becomes redundant and we can just use `INSTALL=${PKG_INSTALL}`. It might be possible to go even further and replace `INSTALL` everywhere with `PKG_INSTALL`.

Reply:
Nice idea, I moved the target logic from the build and install script to `source_package()` and set `PGK_INSTALL` depending on the target.

But `$INSTALL` needs to stay for now, as it's set differently depending on the contexts. E.g. the install scripts copies from `$PKG_INSTALL` to `$INSTALL`.